tests/test_git_history_api.py

```
@@ -0,0 +1,166 @@
+        # 越界参数被钳制
+        resp2 = client.get("/api/git/pushes?page=0&per_page=999")
+        data2 = resp2.get_json()
+        assert data2["page"] == 1
+        assert data2["per_page"] == 100
+        assert len(data2["pushes"]) == 3
+
+    def test_pushes_missing_database(self, client):
+        """registry 未注入 database 时返回显式错误而非通用 500。"""
+        services = app_module.registry._services
+        orig = services.get("database")
+        # 模拟 database 未初始化：临时移除 key
```

Ruff (0.15.17)

[warning] Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)